pkg/api/types.go

```diff
@@ -891,17 +887,32 @@ type ServiceSpec struct {
 	// For hostnames, the user will use a CNAME record (instead of using an A record with the IP)
 	PublicIPs []string `json:"publicIPs,omitempty"`
 
-	// TargetPort is the name or number of the port on the container to direct traffic to.
-	// This is useful if the containers the service points to have multiple open ports.
-	// Optional: If unspecified, the first port on the container will be used.
-	// As of v1beta3 this field will become required in the internal API,
-	// and the versioned APIs must provide a default value.
-	TargetPort util.IntOrString `json:"targetPort,omitempty"`
-
 	// Required: Supports "ClientIP" and "None".  Used to maintain session affinity.
 	SessionAffinity AffinityType `json:"sessionAffinity,omitempty"`
 }
 
+type ServicePort struct {
+	// Optional if only one ServicePort is defined on this service: The
```

Comment:
most of the system won't differentiate whether it is the first port or not,
but validation enforces it.

On Mon, Mar 30, 2015 at 2:04 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6182#discussion_r27431745>
> :
>
> >  	// Required: Supports "ClientIP" and "None".  Used to maintain session affinity.
> >  	SessionAffinity AffinityType `json:"sessionAffinity,omitempty"`
> >  }
> >
> > +type ServicePort struct {
> > +	// Optional if only one ServicePort is defined on this service: The
>
> Technically, isn't the first port allowed to be nameless, regardless how
> many ports there are?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6182/files#r27431745>
> .
>